qa/L0_optuna_bls_model/check_results.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+# Copyright 2021-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import sys
+
+import yaml
+
+
+class TestOutputValidator:
+    """
+    Functions that validate the output
+    of the test
+    """
+
+    def __init__(self, config, test_name, analyzer_log):
+        self._config = config
+        self._models = config["profile_models"]
+        self._analyzer_log = analyzer_log
+
+        check_function = self.__getattribute__(f"check_{test_name}")
+
+        if check_function():
+            sys.exit(0)
+        else:
+            sys.exit(1)
+
+    def check_profile_logs(self):
+        """
+        Check that each model was profiled the number of times
+        corresponding with batch size and concurrency combinations
+
+        (No model config parameter combos expected here!)
+        """
+
+        with open(self._analyzer_log, "r") as f:
+            log_contents = f.read()
+
+        #  Number of configs in search space: 275
+        #  Model - bls:
+        #    concurrency: 1 to 1024 (11)
+        #    instance_group: 1 to 5 (5)
+        #  Composing model - add:
+        #    instance_group: 1 to 5 (5)
+        #
+        #  Minimum number of trials: 14 (5% of search space)
+        #  Maximum number of trials: 28 (10% of search space)
+        #
+        # Then you have 4 x (0-9) for the concurrency sweep on Top 3 + default
```

Comment:
I take it that these can only apply towards the max and not the min? Isn't 4 x 10 + 1 = 41, not 36? The math isn't clear here... tried a few combinations (e.g. top 3 + 1 = 4, not 10)... I can guess at the math, but a bit more clarity here would be helpful. This description is very informative overall though.

Reply:
At maximum there are 4 configs (top-3, default) x 9 concurrencies (not 10, since at least one measurement was already taken per configuration).

I've updated the comments to better reflect this